src/main/java/org/joshsim/lang/interpret/visitor/delegates/JoshFunctionVisitor.java

```diff
@@ -0,0 +1,373 @@
+  }
+
+  /**
+   * Parse a multiple event handler group with conditional branches.
+   *
+   * <p>Process an event handler with multiple implementations selected by conditional branches
+   * (if/elif/else structure).</p>
+   *
+   * @param ctx The ANTLR context from which to parse the multiple event handler group.
+   * @return Fragment containing the event handler group with all conditional branches.
+   */
+  public Fragment visitEventHandlerGroupMultiple(
+      JoshLangParser.EventHandlerGroupMultipleContext ctx) {
+    String fullName = ctx.name.getText();
+    EventKey eventKey = buildEventKey(fullName);
+
+    EventHandlerGroupBuilder groupBuilder = new EventHandlerGroupBuilder();
+    groupBuilder.setEventKey(eventKey);
+
+    int numBranches = ctx.getChildCount() - 1;
+    for (int branchIndex = 0; branchIndex < numBranches; branchIndex++) {
+      int childIndex = branchIndex + 1;
+      Fragment childFragment = ctx.getChild(childIndex).accept(parent);
+
+      if (childFragment.getCompiledSelector().isPresent()) {
+        groupBuilder.addEventHandler(new EventHandler(
+            childFragment.getCompiledCallable(),
+            eventKey.getAttribute(),
+            eventKey.getEvent(),
+            childFragment.getCompiledSelector().get()
+        ));
+      } else {
+        groupBuilder.addEventHandler(new EventHandler(
+            childFragment.getCompiledCallable(),
+            eventKey.getAttribute(),
+            eventKey.getEvent()
+        ));
+      }
+    }
+
+    return new EventHandlerGroupFragment(groupBuilder);
+  }
+
+  /**
+   * Parse a general event handler.
+   *
+   * <p>Process a general event handler and validate that the attribute name is not a reserved
+   * word.</p>
+   *
+   * @param ctx The ANTLR context from which to parse the general event handler.
+   * @return Fragment containing the event handler.
+   * @throws IllegalArgumentException if the attribute name is a reserved word.
+   */
+  public Fragment visitEventHandlerGeneral(JoshLangParser.EventHandlerGeneralContext ctx) {
+    Fragment fragment = ctx.getChild(0).accept(parent);
+    String attributeName = fragment.getEventHandlerGroup().getAttribute();
+    ReservedWordChecker.checkVariableDeclaration(attributeName);
+    return fragment;
+  }
+
+  /**
+   * Checks if a string is a valid event name.
+   *
+   * <p>Determines if the provided string matches one of the predefined event names.</p>
+   *
+   * @param candidate The string to check.
+   * @return true if the string is a valid event name, false otherwise.
+   */
+  private boolean isEventName(String candidate) {
+    return switch (candidate) {
+      case "init", "start", "step", "end", "remove", "constant" -> true;
+      default -> false;
+    };
+  }
+
+  /**
+   * Builds an EventKey from a full event name.
+   *
+   * <p>Parses a dot-separated event name into attribute and event components.
+   * If the last component is a valid event name, it's used as the event name.
```

Comment:
If an event name like step is not specified, the constant initialization event step is used.